tests/CalendarTests.cpp

```diff
@@ -0,0 +1,68 @@
+#include <gtest/gtest.h>
+#include <Calender.hpp>
+#include <algorithm>
+
+struct CalendarTests : public ::testing::Test
+{
+    CalendarTests()
+    {
+        calender_.addLesson(User::Group::weekend, "2019-06-15", "16:30", "Environment configuration" );
+        calender_.addLesson(User::Group::evening, "2019-07-06", "16:30", "Programming tools" );
+        calender_.addLesson(User::Group::evening, "2019-07-07", "16:30", "Basics of C++" );
+    }
+
+    Calender calender_;
+};
+
+TEST_F(CalendarTests, can_add_and_view_Lesson)
+{
+    const auto& result = calender_.viewLesson(User::Group::evening, "2019-07-07", "16:30");
+    EXPECT_EQ(result.getDate(), "2019-07-07");
+    EXPECT_EQ(result.getTime(),  "16:30");
+    EXPECT_EQ(result.getSubject(), "Basics of C++");
+}
+
+TEST_F(CalendarTests, can_add_date_with_dot_slash_minus)
+{
+    calender_.addLesson(User::Group::evening, "2019/06/17", "16:30", "Environment configuration 2" );
+    calender_.addLesson(User::Group::evening, "2019.06.18", "16:30", "Environment configuration 3" );
+
+    const auto& result1 = calender_.viewLesson(User::Group::evening, "2019/06/17", "16:30");
+    const auto& result2 = calender_.viewLesson(User::Group::evening, "2019.06.18", "16:30");
+
+    ASSERT_EQ(result1.getDate(), "2019/06/17");
+    ASSERT_EQ(result2.getDate(), "2019.06.18");
```

Comment:
can we compare dates with different format?

Reply:
Not yet.

Implementation in progress.

Reply:
Why not to restrict to only one format? No implementation will be needed.

Reply:
as per request, only one format will be available